eval/compile.rkt

```diff
@@ -204,9 +209,10 @@
   ; Translates programs into an instruction sequence of operations
   (define (munge prog)
     (define node ; This compiles to the register machine
-      (match (optimize prog)
-        [(list op args ...) (cons op (map munge args))]
-        [_ prog]))
+      (let ([prog* (optimize prog)])
+        (match prog*
+          [(list op args ...) (cons op (map munge args))]
+          [_ prog*])))
```

Comment:
Before, the rule `(exp (log x))` was not working properly because `(optimize prog)` was returning just `x`, which is not a list. Therefore it was falling under `[_ prog]` - where `prog` is the original `(exp (log x))`. As a result optimization was not working